correlation/src/main/java/com/cerner/beadledom/correlation/CorrelationIdContext.java

```diff
@@ -0,0 +1,51 @@
+package com.cerner.beadledom.correlation;
+
+import java.io.Closeable;
+import javax.annotation.Nullable;
+
+/**
+ * A context that provides access to the Correlation ID.
+ */
+public interface CorrelationIdContext {
+  /**
+   * Returns the current Correlation ID, {@code null} if no Correlation ID is in context.
+   */
+  @Nullable
+  String get();
+
+  /**
+   * Sets the Correlation ID. If {@code correlationId} is null, a correlation id is generated.
+   */
+  void set(@Nullable String correlationId);
+
+  /**
+   * Sets the Correlation ID. If {@code correlationId} is null, a Correlation ID is generated. The
+   * returned {@link Closeable} object will clear the Correlation ID when {@link Closeable#close()}
+   * is called. This method is useful to use with try-with-resources when setting a Correlation ID
+   * for the duration of the try block.
+   */
+  CorrelationIdCloseable setCloseable(@Nullable String correlationId);
```

Comment:
I'm not following the purpose of this method. Would this be better served with a method like `withId` that took a lambda?

Reply:
Yeah, it would work better. db80eb4b